src/couch_replicator_doc_processor.erl

```diff
@@ -154,8 +173,9 @@ start_link() ->
 
 
 init([]) ->
+    ok = config:listen_for_changes(?MODULE, nil),
```

Comment:
I think we're optimising prematurely here, it would be simpler to call compat_mode() whenever we need to, since that's comparatively infrequently. We will also be removing this switch in the future.